File: electron_node/docs/AGGREGATOR/test_runner.py
#!/usr/bin/env python3
"""
Run aggregator decision tests using aggregator_decision.py.

Usage:
  python test_runner.py test_vectors.json
"""
import json
import sys
from aggregator_decision import (
    UtteranceInfo, LangProbs,
    decide_stream_action,
)

def load_u(d):
    return UtteranceInfo(
        text=d["text"],
        start_ms=int(d["start_ms"]),
        end_ms=int(d["end_ms"]),
        lang=LangProbs(**d["lang"]),
        quality_score=d.get("quality_score"),
        is_final=bool(d.get("is_final", False)),
        is_manual_cut=bool(d.get("is_manual_cut", False)),
    )

def main():
    path = sys.argv[1] if len(sys.argv) > 1 else "test_vectors.json"
    data = json.load(open(path, "r", encoding="utf-8"))
    ok = 0
    for case in data:
        prev = load_u(case["prev"]) if case.get("prev") else None
        curr = load_u(case["curr"])
        mode = case["mode"]
        got = decide_stream_action(prev, curr, mode)
        exp = case.get("expected_action")
        passed = (exp is None) or (got == exp)
        print(f'{case["id"]}: got={got} expected={exp} -> {"PASS" if passed else "FAIL"}')
        if passed:
            ok += 1
    total = len(data)
    print(f"\\nSummary: {ok}/{total} passed")
    if ok != total:
        sys.exit(1)

if __name__ == "__main__":
    main()

File: electron_node/docs/AGGREGATOR/test_runner.ts
/**
 * Run aggregator decision tests using aggregator_decision.ts
 *
 * Usage:
 *   npx ts-node test_runner.ts test_vectors.json
 */
import * as fs from "fs";
import { decideStreamAction, UtteranceInfo } from "./aggregator_decision";

function loadU(d: any): UtteranceInfo {
  return {
    text: d.text,
    startMs: Number(d.start_ms),
    endMs: Number(d.end_ms),
    lang: d.lang,
    qualityScore: d.quality_score,
    isFinal: Boolean(d.is_final),
    isManualCut: Boolean(d.is_manual_cut),
  };
}

const path = process.argv[2] || "test_vectors.json";
const data = JSON.parse(fs.readFileSync(path, "utf-8"));

let ok = 0;
for (const c of data) {
  const prev = c.prev ? loadU(c.prev) : null;
  const curr = loadU(c.curr);
  const got = decideStreamAction(prev, curr, c.mode);
  const exp = c.expected_action ?? null;
  const pass = exp === null || got === exp;
  console.log(`${c.id}: got=${got} expected=${exp} -> ${pass ? "PASS" : "FAIL"}`);
  if (pass) ok++;
}
console.log(`\nSummary: ${ok}/${data.length} passed`);
if (ok !== data.length) process.exit(1);
